Modules/_datetimemodule.c

```diff
@@ -1837,6 +1837,9 @@ wrap_strftime(PyObject *object, PyObject *format, PyObject *timetuple,
     PyObject *colonzreplacement = NULL; /* py string, replacement for %:z */
     PyObject *Zreplacement = NULL;      /* py string, replacement for %Z */
     PyObject *freplacement = NULL;      /* py string, replacement for %f */
+#ifdef NORMALIZE_CENTURY
+    char year_formatted[5];             /* formatted year with century for %Y */
```

Comment:
This is not enough if the year is negative or has more than 4 digits. Currently the range for years is limited, so this is impossible, but in future we may loosen restrictions and forget to update this line.

Reply:
Right. I've increased the size of the buffer to allow for the lengthiest possible year of -2147481748 permitted by a long int then.